TheBook/GraphTheory/Ramsey.lean

```diff
@@ -0,0 +1,382 @@
+import Mathlib.Combinatorics.SimpleGraph.Clique
+import TheBook.ToMathlib.IndependentSet
+import Mathlib.Tactic.Linarith
+import Mathlib.Data.Nat.Choose.Sum
+
+open SimpleGraph Finset Fintype Nat
+
+-- mathlib?
+lemma Nat.sInf_le_sInf {A B : Set ℕ} (nea : A.Nonempty) (subs : A ⊆ B) : sInf B ≤ sInf A := by
+  by_contra infle
+  exact infle (Nat.sInf_le (Set.mem_of_subset_of_mem subs (sInf_mem nea)))
```

Comment:
Glaube das ist einfach `csInf_le_csInf (OrderBot.bddBelow B) nea subs`? Wobei mir nicht ganz klar ist, warum `A.Nonempty` benötigt wird, normalerweise definiert man einfach `inf ∅ = -∞`.

Reply:
Du verwendest es glaube ich auch später gar nicht?

Reply:
leider nicht: https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Nat/Lattice.html#Nat.sInf_empty
```
theorem sInf_empty : sInf ∅ = 0 
```

Reply:
Dann wäre es ja aber trotzdem weiterhin wahr ohne die `Nonempty` Annahme. Aber aus irgendeinem Grund wird es in `csInf_le_csInf` mit drin sein...